Content.Server/VendingMachines/VendingMachineSystem.cs

```diff
@@ -384,7 +385,22 @@ private void EjectItem(EntityUid uid, VendingMachineComponent? vendComponent = n
                 return;
             }
 
-            var ent = Spawn(vendComponent.NextItemToEject, Transform(uid).Coordinates);
+            // Default spawn coordinates
+            var spawnCoordinates = Transform(uid).Coordinates;
+
+            //Make sure the wallvends spawn outside of the wall.
+            if (HasComp<WallMountComponent>(uid))
+            {
```

Comment:
you dont need this if you are doing a trycomp directly after it

Reply:
Removed